src/chapters/interfaces-and-polymorphism/given-interfaces.rst

```diff
@@ -0,0 +1,137 @@
+Interfaces In The Wild
+======================
+
+The most immediate situations that you’ll encounter in which you’ll want
+to use interfaces are when working with interfaces and classes that are
+part of Java. Here are just a few.
+
+Comparable<T>
+-------------
+
+**Purpose**: A class implements ``Comparable<T>`` in order to allow
+comparison - in a “greater than” and “less than” sense - to another
+instance of the class. This is a “parameterized” interface, which means
+that when using it you need to specify the class that it will be
+comparing. For example, ``Comparable<Job>`` would compare ``Job``
+objects.
+
+**Important Methods**: ``compareTo(T)``
+
+`Comparable<T>
+Documentation <https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/lang/Comparable.html>`__
+
+Comparator<T>
+-------------
+
+**Purpose**: Compare two objects of a given class. When wanting to
+compare objects from a given class, you can create several different
+``Comparator`` classes to allow different types of comparison and
+ordering. The class that you want to compare objects of *does not*
+implement the interface itself.
+
+**Important Methods**: ``compare(T, T)``
+
+`Comparator<T>
+Documentation <https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/util/Comparator.html>`__
+
+This interface can be used to determine, given two objects of the given
+type, which one is “greater” than the other. It is also used by
+collections such as :ref:`an ArrayList <array-list>`
+to sort its contents with the
+:ref:`sort <arraylistsort>`
+method.
+
+.. admonition:: Note
+
+    For more on the differences between ``Comparator`` and ``Comparable``,
+    see `this
+    article <https://www.javatpoint.com/difference-between-comparable-and-comparator>`__.
+
+Iterable<T>
+-----------
+
+**Purpose**: Enable iteration over a collection of objects using a
+for-each loop
+
+**Important Methods**: ``iterator()``
+
+`Iterable<T>
+Documentation <http://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html>`__
+
+This interface is implemented by the ``ArrayList<T>`` class, which we’ve
+been using throughout this course.
+
+.. admonition:: Example
+
+   .. sourcecode:: java
+      :linenos:
+
+      Iterable<String> collection = new ArrayList<>();
+
+      // Add items to the collection
+
+      for (String item : collection) {
+         // do something with each item
+      }
+
+List<T>
+-------
+
+**Purpose**: Enable access to objects in a collection by index. In other
```

Comment:
"Enables"

Also, "In other words, it enables..."

Reply:
Done! Thanks!